frontend/src/metabase/lib/urls/collections.ts

```diff
@@ -31,7 +31,12 @@ function slugifyPersonalCollection(collection: Collection) {
   return slug;
 }
 
-export function collection(collection?: Collection) {
+export function collection(
+  collection?: Pick<Collection, "id" | "name"> & {
```

Comment:
Let me know if there's a better way to do this typing - I can't do 
```
Pick<Collection, "id" | "name" | "personal_owner_id" | "originalName">
```

since we want the last two properties to be optional. Maybe we can do a `Partial<Pick<<Collection | [...]>>` but idk how much nicer that would be

Reply:
🤦  my IDE wasn't directing me to that definition, lovely.

In any case, I think the idea is that we have a ton of different permutations of the `Collection` object (i.e. `Pick`s, `Partial`s, the one in `Search` etc), and we'll sometimes want to get the collection URL from whatever object that is. So, if we try and pass a partial/picked version of the `Collection` then we'll get a type error if we're typed with the full `Collection`

Reply:
Ah yeah that's what I was referring to. I was answering the `Can't we just use Collection from metabase-types/api without redefining it in line 9?` 😄